demo-project/demo-api.raml

```diff
@@ -0,0 +1,88 @@
+#%RAML 1.0
+---
+title: Demo API
+baseUri: http://localhost:8080/
+
+/values:
+    get:
+      description: "Retrieve an array of value entries"
+      responses:
+        200:
+          body:
+            application/json:
+              example: |
+                [
+                    {
+                        "key": "k01",
+                        "value": "some value",
+                        "id": 1
+                    },
+                    {
+                        "key": "k02",
+                        "value": "another value",
+                        "id": 2
+                    }
+                ]
+    post:
+        description: "Create a new value entry"
+        body:
+          application/json:
+            type: value
+            example: |
+              {
+                  "key": "k0",
+                  "value": "a value",
+                  "id": null
+              }
+        responses:
+          201:
+            body:
+              application/json:
+                  example: |
+                      {
+                        "key": "k0",
+                        "value": "a value",
+                        "id": 1
+                      }
+            headers:
+                  Location:
+                    example: "http://localhost:8080/values/1"
+    /{id}:
+        put:
+            body:
+              application/json:
+                type: value
+                example: |
+                  {
+                      "key": "k0",
+                      "value": "a value",
+                      "id": null
```

Comment:
I don't recommend mentioning `id` on the body of the PUT, since it's already in the URI path.